src/components/navigation-bar/index.js

```diff
@@ -0,0 +1,130 @@
+import BaseComponent from "../base-component.js";
+// NOTE: this import needed for bootstrap navigation toaster icon
+import { Offcanvas } from "bootstrap";
+import Modal from "../modal/index.js";
+import LoginForm from "../login-form/index.js";
+import { isAuthorized } from "../../router/guards/index.js";
+
+export default class NavigationBar extends BaseComponent {
+  links = [
+    { name: "Home", path: "/home", guard: () => true },
+    {
+      name: "Create product",
+      path: "/create-product",
+      guard: () => isAuthorized(),
+    },
+    { name: "Orders", path: "/orders", guard: () => isAuthorized() },
+  ];
+
+  buttons = [
+    { name: "Login", guard: () => !isAuthorized() },
+    { name: "Logout", guard: () => isAuthorized() },
+  ];
+
+  constructor() {
+    super();
+    this.init();
+    this.update();
+  }
+
+  createLinks() {
+    const fragment = new DocumentFragment();
+
+    for (const link of this.links) {
+      const li = document.createElement("li");
+
+      if (!link.guard()) {
+        continue;
+      }
+
+      li.innerHTML = `
+        <li class="nav-item">
+          <a class="nav-link" aria-current="page" href="${link.path}">${link.name}</a> 
+        </li> 
+      `;
+
+      fragment.append(li);
+    }
+
+    this.subElements.navList.append(fragment);
+  }
+
+  createBtns() {
+    const fragment = new DocumentFragment();
+
+    for (const button of this.buttons) {
+      const wrapper = document.createElement("div");
+
+      if (!button.guard()) {
+        continue;
+      }
+
+      wrapper.innerHTML = `
+        <li class="nav-item">
+          <button type="button" 
+                  class="btn btn-link" 
+                  data-element="${button.name.toLowerCase()}Btn" 
+                  data-cy="${button.name.toLowerCase()}-btn">
+            ${button.name}
+          </button>
+        </li>
+      `;
+
+      fragment.append(wrapper.firstElementChild);
+    }
+
+    this.subElements.navList.append(fragment);
+  }
+
+  update() {
+    this.subElements.navList.innerHTML = "";
+    this.createLinks();
+    this.createBtns();
+    this.getSubElements();
+    this.initEventListeners();
```

Comment:
**Performance Issue**: The 'update' method clears and recreates all navigation links and buttons every time it is called, which is inefficient and can lead to poor performance on frequent updates. <br> **Fix**: Improve the efficiency of the 'update' method by checking if updates are necessary before clearing and recreating elements, or by updating only the changed parts of the DOM. <br> **Code Suggestion**: 
 ```
 +    update() {
 +      if (needsUpdate(this.links, this.buttons)) {
 +        this.subElements.navList.innerHTML = "";
 +        this.createLinks();
 +        this.createBtns();
 +        this.getSubElements();
 +        this.initEventListeners();
 +      }
 +    }
 ```